### DISCUSSION TITLE
QuestPDF Unity integration

### DISCUSSION THREAD

**Discussion by FlorianSchmittDE** [QuestPDF/QuestPDF#325] [Q&A] [answered]
Dear QuestPDF community,

do any of you have experience integrating QuestPDF into Unity and got it to work on iOS?
I can use the api in the project but when i run it the skia libs can't be found. There is an old tutorial that mentions that the skialib dylib can be renamed to bundle. Unfortunately i'm stuck at this point. Thanks in advance

**Comment by gunseliozkan** [comment]
I am in exact same situation. Any help would be appreciated!

**Comment by Charity29614** [comment, marked as answer]
Since QuestPDF is a .NET-based library and doesn’t work natively on iOS within Unity due to its dependency on native SkiaSharp/Skia libraries, you should offload PDF generation to a backend server.

Create a .NET console app with QuestPDF. 

Build it for macOS or windows. 

Call from Node.js server to communicate with the console app to generate the pdf. 

Then on the IOS build: Make HTTP request to /pdf and download the result.

This bypasses Skia issues on iOS by handling PDF generation server-side.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
